src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -21,99 +19,1454 @@ const ActorInit En_Mb_InitVars = {
+    }
+    if (this->unk_328 != 0) {
+        switch (limbIndex) {
+            case 7:
+                phi_v1 = 0;
+                break;
+            case 12:
+                phi_v1 = 1;
+                break;
+            case 17:
+                phi_v1 = 2;
+                break;
+            case 9:
+                phi_v1 = 3;
+                break;
+            case 14:
+                phi_v1 = 4;
+                break;
+            case 6:
+                phi_v1 = 5;
+                break;
+            case 20:
+                phi_v1 = 6;
+                break;
+            case 25:
+                phi_v1 = 7;
+                break;
+            case 22:
+                phi_v1 = 8;
+                break;
+            case 27:
+                phi_v1 = 9;
+                break;
+        }
+        if (phi_v1 >= 0) {
+            Matrix_MultVec3f(&D_80AA9E14, &sp24);
+            this->unk_14C[phi_v1].x = sp24.x;
+            this->unk_14C[phi_v1].y = sp24.y;
+            this->unk_14C[phi_v1].z = sp24.z;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/func_80AA9870.s")
+void EnMb_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Vec3f D_80AA9E20[] = {
+        { 4000.0f, 7000.0f, 3500.0f },
+        { 4000.0f, 0.0f, 3500.0f },
+        { -4000.0f, 7000.0f, 3500.0f },
+    };
+    static Vec3f D_80AA9E44[] = {
+        { -4000.0f, 7000.0f, 3500.0f },
+        { -4000.0f, 0.0f, 3500.0f },
+        { 4000.0f, 0.0f, 3500.0f },
+    };
+    s32 i;
+    f32 phi_f0;
+    Vec3f phi_s2[3];
+    Vec3f phi_s0[3];
+    s32 tempPosIndex;
+    EnMb* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/EnMb_Draw.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount, 0,
+                     EnMb_PostLimbDraw, thisx);
+    if (thisx->params != 0) {
+        if (this->attackParams > 0) {
+            func_80AA92B8(thisx, globalCtx);
+        }
+        for (i = 0; i < 3; i++) {
+            Matrix_MultVec3f(&D_80AA9E20[i], &phi_s2[i]);
+            Matrix_MultVec3f(&D_80AA9E44[i], &phi_s0[i]);
+        }
+        func_800627A0(&this->collider3, 0, &phi_s2[0], &phi_s2[1], &phi_s2[2]);
+        func_800627A0(&this->collider3, 1, &phi_s0[0], &phi_s0[1], &phi_s0[2]);
+    }
+    if (this->unk_328 != 0) {
+        thisx->dmgEffectTimer += 1;
+        if (this->unk_328 >= 0) {
+            this->unk_328--;
+        }
+        if ((this->unk_328 & 3) == 0) {
+            phi_f0 = 2.5f;
+            if (thisx->params == 0) {
+                phi_f0 = 4.0f;
+            }
+            tempPosIndex = this->unk_328 >> 2;
+            EffectSsEnIce_SpawnFlyingVec3s(globalCtx, thisx, &this->unk_14C[tempPosIndex], 0x96, 0x96, 0x96, 0xFA, 0xEB,
+                                           0xF5, 0xFF, phi_f0);
+        }
```

Comment:
these aregs can prob be in decimal and i imagine you can name them based on the args from the prototype of the effect function